sendgrid/message.py

```diff
@@ -146,10 +163,19 @@ def add_content_id(self, cid, value):
         self.content[cid] = value
 
     def set_headers(self, headers):
+        # if not headers:
+        #     self.headers = headers
+        # else:
+        # Convert to object to be able in order to append to it
+        if isinstance(self.headers, str):
+            self.headers = json.loads(self.headers)
         if isinstance(headers, str):
-            self.headers = headers
-        else:
-            self.headers = json.dumps(headers)
+            headers = json.loads(headers)
+        for key, value in headers.iteritems():
+            self.headers[key] = value
+
+        # Convert back to string
+        self.headers = json.dumps(self.headers)

```

Comment:
@tgwizard The reason I've had to to do the conversion is so as to not break backwards compatibility.